Source/JavaScriptCore/builtins/JSIteratorConstructor.js

```diff
@@ -71,3 +71,44 @@ function from(value)
     // 6. Return wrapper.
     return @wrapForValidIteratorCreate(iterator, iteratorNextMethod);
 }
+
+// https://tc39.es/proposal-iterator-sequencing/#sec-iterator.concat
+function concat(/* ...items */)
+{
+    "use strict";
+    var len = @argumentCount();
+
+    var iterables = [];
+    for (var i = 0; i < @argumentCount(); ++i) {
+        var iterable = arguments[i];
+        if (!@isObject(iterable))
+            @throwTypeError("Iterator.concat expects all arguments to be objects");
+
+        var openMethod = iterable.@@iterator;
+        if (@isUndefinedOrNull(openMethod))
```

Comment:
According to the spec,

> c. If method is undefined, throw a TypeError exception.

so we should check only `undefined`? (Or feedback to the spec)

Reply:
i was matching what was done for some of the other iterator helper methods, so if it truly is that we only want to check for `undefined` then we probably need to update those too (or i could be incorrectly reading the spec for those methods)

Reply:
GetMethod returns `undefined` for both `null` and `undefined`. See https://tc39.es/ecma262/multipage/abstract-operations.html#sec-getmethod.

It also checks for callability and throws if the value isn't callable.